File: src/main/js/y2020/Day6.js
const _ = require('lodash');
const fs = require('fs');

const input = fs.readFileSync('../../inputs/2020/6.txt', 'utf8').split('\n\n');

const sum1 = _.sum(input.map((group) => {
  return _.uniq(
    group
      .split('\n')
      .flatMap(x => x.split(""))
  ).length
}));

const sum2 = _.sum(input.map((group) => {
  return _.intersection(
    ...group
      .split('\n')
      .map(x => x.split(""))
  ).length
}));

console.log(sum1) //  6530
console.log(sum2) //  3323

File: src/main/go/day6.go
package main

import (
	"fmt"
	"strings"
)

func day6() {
	input := strings.Split(MustReadFile("../inputs/2020/6.txt"), "\n\n")

	union := 0
	intersection := 0

	for _, group := range input {
		charCounts := make(map[string]int)

		members := strings.Split(group, "\n")
		for _, member := range members {
			for _, answer := range strings.Split(member, "") {
				charCounts[answer] = charCounts[answer] + 1
			}
		}

		union += len(charCounts)
		for _, count := range charCounts {
			if count == len(members) {
				intersection++
			}
		}
	}

	fmt.Println(union)
	fmt.Println(intersection)

}
